lib/improver/nowcasting/optical_flow.py

```diff
@@ -345,11 +351,22 @@ def __init__(self, data_smoothing_radius=7., data_smoothing_method='box',
         self.data2 = None
         self.shape = None
 
+    def __repr__(self):
+        """Represent the plugin instance as a string."""
+        result = ('<OpticalFlow: data_smoothing_radius_km: {}, '
+                  'data_smoothing_method: {}, boxsize_km: {}, '
+                  'iterations: {}, point_weight: {}>')
+        return result.format(
+            self.data_smoothing_radius_km, self.data_smoothing_method,
+            self.boxsize_km, self.iterations, self.point_weight)
+
     @staticmethod
-    def corner(data, axis=None):
+    def interp_to_midpoint(data, axis=None):
         """
-        Calculates the average of four corner points at each point on a grid.
-        If axis is not None, only averages over the spatial axis specified.
+        Interpolates the average of four corner points at each grid point onto
+        a new grid of reduced dimensions.  If axis is not None, only averages
+        over the spatial axis specified.  If the input array has an axis of
+        length 1, this function returns an empty array: [].
```

Comment:
I think this can be improved a bit more. "Interpolates to the x-y mid-point resulting in a new grid with dimensions reduced in length by one."